Source/Backlang.Codeanalysis/Parsing/AST/Statements/SwitchStatement.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Loyc.Syntax;
+
+namespace Backlang.Codeanalysis.Parsing.AST.Statements;
+public sealed class SwitchStatement : IParsePoint<LNode>
+{
+
+    /*
+     * switch element {
+     *  case cond: oneExpr;
+     *  case cond: { block; }
+     *  if boolean: oneExpr;
+     *  if boolean: { block; }
+     *  default: oneExpr;
+     *  default: { block; }
+     * }
+     */
+    public static LNode Parse(TokenIterator iterator, Parser parser)
+    {
+        var element = Expression.Parse(parser);
+
+        parser.Iterator.Match(TokenType.OpenCurly);
+
+        var cases = new LNodeList();
+
+        while (!parser.Iterator.IsMatch(TokenType.CloseCurly))
+        {
+            if(iterator.IsMatch(TokenType.Case))
+                cases.Add(ParseCase(parser));
+            if (iterator.IsMatch(TokenType.If))
+                cases.Add(ParseIf(parser));
+            if (iterator.IsMatch(TokenType.Default))
+                cases.Add(ParseDefault(parser));
+        }
+
+        parser.Iterator.Match(TokenType.CloseCurly);
+
+        return SyntaxTree.Switch(element, cases);
+    }
+
+    private static LNode ParseCase(Parser parser)
+    {
+        parser.Iterator.Match(TokenType.Case);
+
+        var condition = Expression.Parse(parser);
```

Comment:
This allows parsing: case isAlpha(mychar)
is this considered?

Reply:
yes this is. when the value that comes out of the function is equal to the element of the switch statement, the code of this case should get executed